openpype/hosts/houdini/plugins/create/create_bgeo.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+"""Creator plugin for creating pointcache bgeo files."""
+from openpype.hosts.houdini.api import plugin
+from openpype.pipeline import CreatedInstance
+from openpype.lib import EnumDef
+
+
+class CreateBGEO(plugin.HoudiniCreator):
+    """BGEO pointcache creator."""
+    identifier = "io.openpype.creators.houdini.bgeo"
+    label = "BGEO PointCache"
+    family = "bgeo"
```

Comment:
Out of scope for this PR but I was testing the Houdini publisher when I got confused (again, because I remember coming across this same confusion a couple weeks back) trying to find an Alembic cache family and realizing that it was the `Point Cache`. However, that got me thinking and made me remember this PR where you were creating a new family and I'm noticing there's quite semantic divergence on how you have been creating the more recent families and I don't understand why we are not making use of `representations` instead of creating new families for each file format.
Wouldn't it make more sense to have a single `Point Cache` family that allows us to define which representations we want for it (i.e., `Abc`, `Bgeo`, `ASS`, `USD`...) so it's up to the project settings to choose what data formats we want our families to be on? Same applies to `Model`, `Camera`... In my prior studio we published the different file formats as `components` on the same ftrack asset version, so sometimes you'd prefer to import the bgeo representation but some others the alembic cache (same for textures to store .exr and .tx or shaders for .mtlx, .ass...). Of course there will be occasions where there might be some slight differences of what's stored on the representation due to the missmatch of features but I think there's some big limitations on the alternative. I wouldn't want the artists to select every family we choose to support on the project, I would rather them just needing to know they have to publish a "Cache" and in our settings I could choose to not only create a `Bgeo` cache but perhaps a `USD` one as well, that way I could slowly migrate to a different format. Does this make any sense or I need some extra coffee?

Reply:
> Wouldn't it make more sense to have a single Point Cache family that allows us to define which representations we want for it (i.e., Abc, Bgeo, ASS, USD...)

It's not **that** trivial due to them actually being separate ROP nodes (~and also requiring different validators to trigger in production~). So the complexity of maintaining that increases and the flexibility of the individual nodes somewhat go down I'd say. It's not impossible however but this would definitely require a custom HDA to make it maintainable.

I'm not entirely sure however about writing say `bgeo` + `alembic` + `usd` family into a single subset. Again, it's possible - just add the different families into a single instance and it does it. But does OpenPype / Ayon allow changing the family of an existing subset midway, like remove an `alembic` family and add a `bgeo` if the next version doesn't have it. Sounds dangerous if we start wiggling that around and someone starts ticking off and on a specific "output family" for the instance when publishing new versions?

Older iterations of OpenPype / Avalon had `families` on versions so they were per version, but at some point they were moved to Subsets to make them more static with a reason - to avoid the expected data to be versioned to to suddenly be something different. E.g. latest version 8 not having a USD file, so to what version do you update then when updating the loaded USD? Do you report an error? Take the last existing? etc. Complexity does quickly increase.

Another side note - for e.g. USD you tend to set a different "path" LOPs path instead of SOP path for Alembic, etc. So even setting what you're outputting could differ?

Reply:
Thanks @BigRoy for the quick response and explanation!! I definitely agree that the complexity would increase but knowing how many times on my prior studio we were able to fix production needs by having the flexibility of easily swapping representations/components of any existing asset version makes me worry whether that's something that could become a blocker with the current design.

Are the key concepts definitions from the OP docs outdated? I'm a bit confused when you mention "`bgeo` + `alembic` + `usd` **family** into a single subset". Wouldn't that be "`bgeo` + `alembic` + `usd` **representations** into a single subset"? Reading that glossary again (https://openpype.io/docs/artist_concepts#representation) this seems to describe exactly how I had envisioned the design to be, you choose the family, the subset and then the representations of that subset (family > subset > representation). But from what I'm seeing, "representations" aren't really used here, mostly just for storing a thumbnail of the same subset?

I think that creating custom HDAs might be a requirement to scale the OP integration and make it more Houdini native in the long run (allowing artists to create node networks directly and setting any dependencies for doing publishes instead of needing to go through the UIs, although that would still be possible). On my prior studio we had a system that hijacked the `hou.Node` python class by inserting it as an inherit into our custom python classes so you could write any python code and you could call it for any HDA or Houdini node directly from the node's python object (i.e., `hou.node("custom_node").custom_function()`, `hou.node("alembic_node").custom_function()`), not needing to write the code within the HDA. We also had custom nodes with very simple purposes (i.e., "OP Publisher", "OP Cache") so for TDs (and even the developers to provide new publishers) it made it very easy to build more complex functionality by combining the smaller nodes (think of Bifrost Compounds or Nuke gizmos). 

As for your worry about "representations" missing across different versions, the way we did it was very simple, if you try to update to a newer version that doesn't have the representation you are using, it just errors out saying that the "representation" doesn't exist. However, that was very uncommon as most artists didn't mess with the default job settings of which "representation/component" should be published by family. On the other hand, imagine the alternative situation where I might want to have "Bgeo" and "Alembic" caches throughout the entire production. How do you currently keep those separate families "on sync" with this design? If they don't **always** publish both, the versions get out of sync and all of a sudden it makes it very hard to know which family "represents" the other family. I think that's where "representations" should really be used for?

Anyway, I'm still very new to OpenPype and I still need to fight a lot more with the Houdini OP implementation to get all of these concepts wrapped in my brain, I might change my mind again then haha

Definitely agree to start a GH issue/discussion to discuss this further.